shell_integration/windows/OCUtil/RemotePathChecker.cpp

```diff
@@ -172,10 +176,13 @@ RemotePathChecker::~RemotePathChecker()
     CloseHandle(_newQueries);
 }
 
-vector<wstring> RemotePathChecker::WatchedDirectories()
+void RemotePathChecker::UpdateWatchedDirectories(vector<wstring>& localWatchedDirectories, unsigned& localVersion) const
 {
+    if (localVersion == _watchedDirectoriesVersion)
```

Comment:
After realizing that multiple explorer windows will call `IsMemberOf` concurrently, I've put the vector in a shared_ptr and each call will keep a ref to that copy on the stack (@ogoffart's idea of course).